File: Week-5/leetcode/valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        li = []
        for i in range(len(s)):
            if s[i]=='(' or s[i]=='{' or s[i]=='[':
                li.append(s[i])
            elif s[i]==')' and li and li[-1]=='(':
                li.pop()
            elif s[i]=='}'and li and li[-1]=='{':
                li.pop()
            elif s[i]==']' and li and li[-1]=='[':
                li.pop()
            else:
                return False

        return True if not li else False

File: Week-5/leetcode/valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        //if (s.length()<2) return false;
        stack<char> stk;
        for(int i=0; i<s.length(); i++){
            if(s[i]=='('||s[i]=='{'||s[i]=='['){
                stk.push(s[i]);
            }
            else if(s[i]==')' && !stk.empty() && stk.top()=='('){
                stk.pop();
            }else if(s[i]=='}' && !stk.empty() && stk.top()=='{'){
                stk.pop();
            }else if(s[i]==']' && !stk.empty() && stk.top()=='['){
                stk.pop();
            }else return false;
        }
        return stk.empty();
    }
};